lib/file-source/src/file_watcher.rs

```diff
@@ -51,23 +53,35 @@ impl FileWatcher {
             rdr.seek(io::SeekFrom::Start(file_position)).unwrap()
         };
 
+        let reader = Self::wrap_reader_for_filetype(rdr, &path);
+
         Ok(FileWatcher {
             path: path,
             findable: true,
-            reader: rdr,
+            reader: reader,
             file_position: file_position,
             devno: metadata.dev(),
             inode: metadata.ino(),
             is_dead: false,
         })
     }
 
+    // Depending on the file type we create a different type of reader.
+    // This allows us to handle compressed files as well as uncompressed files.
+    fn wrap_reader_for_filetype(reader: io::BufReader<File>, path: &PathBuf) -> Box<dyn BufRead> {
+        match path.extension().and_then(OsStr::to_str) {
+            Some("gzip") => Box::new(io::BufReader::new(MultiGzDecoder::new(reader))),
```

Comment:
Probably want this to be `Some("gzip") | Some("gz") => ...`

Reply:
True. Changed. 👍